bin/samtools_depth_average_coverage.py

```diff
@@ -1,27 +1,71 @@
 #!/usr/bin/env python3
 """
-Script for finding the average coverage of each scaffold in a SAMtools depth output file
+Script for finding the average coverage of each scaffold in a SAMtools depth output file.
+originally written by Eerik Aunin (ea10)
+re-wriiten by Yumi Sims (yy5)
 """
 
 import sys
+import argparse
 import general_purpose_functions as gpf
 
-in_path = sys.argv[1]
-in_data = gpf.ll(in_path)
-scaffs_dict = dict()
-
-for line in in_data:
-    split_line = line.split()
-    scaff_name = split_line[0]
-    coverage = int(split_line[2])
-    if scaff_name in scaffs_dict:
-        scaffs_dict[scaff_name]["coverage_sum"] += coverage
-        scaffs_dict[scaff_name]["scaff_len"] += 1
-    else:
-        scaffs_dict[scaff_name] = dict()
-        scaffs_dict[scaff_name]["coverage_sum"] = coverage
-        scaffs_dict[scaff_name]["scaff_len"] = 1
-
-for scaff_name in scaffs_dict:
-    average_coverage = scaffs_dict[scaff_name]["coverage_sum"] / scaffs_dict[scaff_name]["scaff_len"]
-    print(scaff_name + "," + str(average_coverage))
+# Define the version number
+__version__ = "1.0.0"
```

Comment:
Maybe replace the version and `display_version` with `parser.add_argument("--version", action="version", version="1.0.0")`

Reply:
I did that in different code, forgot this one, will change, thanks.